### PR TITLE
DSL2: Add Mtnucratio

### LINKED ISSUE #955: DSL2: Add mtnuclear ratio
Add mt nuclear ratio module

### PR DESCRIPTION
**Pull request by jbv2**
Closes #955

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/eager/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/eager _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CITATIONS.md [hunk 1]
```diff
@@ -40,6 +40,9 @@
 
   > Peltzer, A., Jäger, G., Herbig, A., Seitz, A., Kniep, C., Krause, J., & Nieselt, K. (2016). EAGER: efficient ancient genome reconstruction. Genome Biology, 17(1), 1–14. doi: [10.1186/s13059-016-0918-z](https://doi.org/10.1186/s13059-016-0918-z)
 
+  - MTNucRatioCalculator (Unpublished)
```
Comment (jfy133) [inline CITATIONS.md:43]:
```suggestion

  - MTNucRatioCalculator (Unpublished)
```

I've asked Alex Peltzer if he can get a Zenodo DOI for us

File: conf/modules.config [hunk 2]
```diff
@@ -412,4 +412,12 @@ process {
             pattern: '*.flagstat'
         ]
     }
+
+    withName: MTNUCRATIO {
```
Comment (jfy133) [inline conf/modules.config:416]: I don't think we need to publish this, it's only useful in the final MultiQC report (so just need to make sure that is included in the MultiQC config later :) )

File: docs/output.md [hunk 3]
```diff
@@ -193,3 +193,16 @@ Deduplication is carried by two possible tools, as described below. However the
...
+<details markdown="1">
+<summary>Output files</summary>
+
+`nuclear_contamination/`
+
+- `*.{.mtnucratio}`: number of reads on mitochondrium and on on nuclear chromosomes. Givs the mt/nuc Ratio.
+
+</details>
+
+[MTtoNucRatio](https://github.com/apeltzer/MTNucRatioCalculator) is a tool that reports the number of reads aligned to a mitochondrial entry in your reference FASTA to all other entries. This will typically be high but will vary depending on tissue type.
```
Comment (jfy133) [inline docs/output.md:208]: Can be removed (see below)
 
```suggestion
```

File: nextflow.config [hunk 4]
```diff
@@ -114,6 +114,10 @@ params {
     skip_deduplication = false
     deduplication_tool = 'markduplicates'
 
+    // MtNucRatio
+    run_mtnucratio = false
+    mtnucratio_header = 'MT'
```
Comment (jfy133) [inline nextflow.config:119]: CAn you rename this 
```suggestion
    mitochondrion = 'MT'
```

AS thisi s what Iv'e called it in #952 - or we could rename it to `mitochondrion_header`... thoughts @TCLamnidis ?

File: nextflow_schema.json [hunk 5]
```diff
@@ -585,11 +585,34 @@
...
+            "help_text": "> If using TSV input, Mitochondrial to Nuclear Ratio calculation is calculated per deduplicated library (after lane merging)",
+            "fa_icon": "fas fa-balance-scale-left",
+            "properties": {
+                "run_mtnucratio": {
+                    "type": "boolean",
+                    "description": "Turn on mitochondrial to nuclear ratio calculation.",
+                    "help_text": "Turn on the module to estimate the ratio of mitochondrial to nuclear reads.",
+                    "fa_icon": "fas fa-balance-scale-left"
+                },
+                "mtnucratio_header": {
```
Comment (jfy133) [inline nextflow_schema.json:608]: If you update the parameter as I described above, make sure torename here.

**Review by jfy133 — COMMENTED** [review]
overall LGTM, some thought about the parameter name

File: nextflow_schema.json (hunk 5 above)
Comment (jbv2) [reply nextflow_schema.json:608]:
```suggestion
                "mitochondrion_name": {
```

File: nextflow.config (hunk 4 above)
Comment (jbv2) [inline nextflow.config:119]:
```suggestion
    mitochondrion_name = 'MT'
```

**Review by jbv2 — COMMENTED** [review]
Renaming mitochondrion_name parameter

File: CITATIONS.md (hunk 1 above)
Comment (jfy133) [reply CITATIONS.md:47]: But we can merge now anyway while we wait (we will have to wait untilthe weekend)

File: conf/modules.config (hunk 2 above)
Comment (jfy133) [reply conf/modules.config:427]: Yup that's fine :)

File: nextflow.config (hunk 4 above)
Comment (jfy133) [reply nextflow.config:121]: Lets say header to make sure it's linked with the FASTA format, I can update my PR accordingly

File: nextflow.config [hunk 6]
```diff
@@ -116,6 +116,10 @@ params {
...
```
Comment (jbv2) [inline nextflow.config:121]:
```suggestion
    mitochondrion_header = 'MT'
```

File: nextflow_schema.json [hunk 7]
```diff
@@ -612,6 +612,29 @@
...
```
Comment (jbv2) [inline nextflow_schema.json:630]:
```suggestion
                "mitochondrion_header": {
```

**Review by jbv2 — COMMENTED** [review]
Renaming params to mitochondrion_header

**Review by jbv2 — COMMENTED** [review]
Accepting changes

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Review by TCLamnidis — CHANGES_REQUESTED** [review]
The docs of Preseq got deleted. Otherwise LGTM

**Review by jbv2 — COMMENTED** [review]
Adding missing Preseq docs/output

File: nextflow_schema.json [hunk 8]
```diff
@@ -613,6 +613,29 @@
             },
             "fa_icon": "fas fa-clone"
         },
+        "mitochondrial_to_nuclear_ratio": {
+            "title": "Mitochondrial to Nuclear Ratio",
+            "type": "object",
+            "description": "Options for the calculation of ratio of reads to one chromosome/FASTA entry against all others.",
+            "default": "",
+            "help_text": "> If using TSV input, Mitochondrial to Nuclear Ratio calculation is calculated per deduplicated library (after lane merging)",
```
Comment (jfy133) [inline nextflow_schema.json:621]:
```suggestion
            "help_text": "If using TSV input, Mitochondrial to Nuclear Ratio calculation is calculated per deduplicated library (after lane merging)",
```

File: workflows/eager.nf [hunk 9]
```diff
@@ -217,6 +219,21 @@ workflow EAGER {
...
+    // MODULE: MTNUCRATIO
+    //
+    if ( params.run_mtnucratio ) {
+        mtnucratio_input = ch_dedupped_bams
+        .map {
+            meta, bam, bai ->
+            [ meta, bam ]
+        }
+
+        MTNUCRATIO(mtnucratio_input, params.mitochondrion_header)
```
Comment (jfy133) [inline workflows/eager.nf:232]:
```suggestion
        MTNUCRATIO( mtnucratio_input, params.mitochondrion_header )
```

**Review by jfy133 — CHANGES_REQUESTED** [review]
output.md `preseq` file is still delted, is that to be expeted? We've merged PreSeq in now rigjt?

But otherwise some minor formatting things and then I think we should be OK :)

**Comment by TCLamnidis** [comment]
@jfy133 this also needs a CI test parameter added, right?

**Comment by jfy133** [comment]
> @jfy133 this also needs a CI test parameter added, right?

Yes, needs to be turned on in one of the tetss and the header param passed to it in the test config

**Comment by jbv2** [comment]
@nf-core-bot fix linting

**Review by TCLamnidis — APPROVED** [review]

File: nextflow_schema.json [hunk 10]
```diff
@@ -613,6 +613,29 @@
...
+                    "type": "boolean",
+                    "description": "Turn on mitochondrial to nuclear ratio calculation.",
+                    "help_text": "Turn on the module to estimate the ratio of mitochondrial to nuclear reads.",
+                    "fa_icon": "fas fa-balance-scale-left"
+                },
+                "mitochondrion_header": {
+                    "type": "string",
+                    "default": "MT",
+                    "description": "Specify the name of the reference FASTA entry corresponding to the mitochondrial genome (up to the first space).",
+                    "help_text": "Specify the FASTA entry in the reference file specified as --fasta, which acts as the mitochondrial 'chromosome' to base the ratio calculation on. The tool only accepts the first section of the header before the first space. The default chromosome name is based on hs37d5/GrCH37 human reference genome. Default: 'MT'",
```
Comment (jfy133) [inline nextflow_schema.json:634]:
```suggestion
                    "help_text": "Specify the FASTA entry in the reference file specified as --fasta, which acts as the mitochondrial 'chromosome' to base the ratio calculation on. The tool only accepts the first section of the header before the first space. The default chromosome name is based on hs37d5/GrCH37 human reference genome.",
```

**Review by jfy133 — APPROVED** [review]

**Pull request merged**